### PR TITLE
MWPW-147989 and MWPW-151786: handle localized path for quiz spreadsheets and doc files

### PR DESCRIPTION
**Pull request by JackySun9**
* 1. add localized prefix for all paths in quiz and quiz results spreadsheets if need, including quiz, quiz-results, footer fragments
* 2. add localized prefix for all paths in quiz and quiz results dic files if need, including quiz, quiz-results
* 3. add unit tests
* 4. so the logic is if you already add localized prefix, the code will skip it, if not and not en-US, the code will add localized prefix automatically

Resolves: [MWPW-147989](https://jira.corp.adobe.com/browse/MWPW-147989)
[MWPW-151786](https://jira.corp.adobe.com/browse/MWPW-151786)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/?martech=off
- After: https://stage--milo--JackySun9.hlx.page/?martech=off

How to test?
Test with ES, it is using all paths without country code:
URL: https://stage--cc--adobecom.hlx.page/es/creativecloud/plan-recommender/quiz2?milolibs=stage--milo--JackySun9
You can see https://stage--cc--adobecom.hlx.page/es/creativecloud/plan-recommender/quiz2 is loading English content,
Screenshot 2024-06-04 at 11 19 54 AM

but with updated code it is loading ES content, also we set all fragment paths in LocalStorage
Screenshot 2024-06-04 at 11 19 46 AM
Screenshot 2024-06-04 at 11 19 29 AM

Test with DE, it is using all paths with country code:
URL: https://stage--cc--adobecom.hlx.page/de/creativecloud/plan-recommender/quiz2?milolibs=stage--milo--JackySun9

### CODE REVIEW

**Review by colloyd — APPROVED** [review]
Looks and works great. The addition of the getLocalilzed URL in utils.js is great and the shift of fragment url localization to quiz, rather than quiz-results is excellent.

File: libs/blocks/quiz/utils.js [hunk 1]
```diff
@@ -497,3 +498,17 @@ export const getAnalyticsDataForLocalStorage = (config) => {
...
+  const prefix = locale?.prefix;
+  const localCode = locale?.ietf ?? 'en-US';
+
+  let localizedURL = originalURL;
+
+  if (localCode !== 'en-US' && !originalURL.startsWith(`${prefix}/`)) {
+    localizedURL = `${prefix}${originalURL}`;
+  }
+
+  return localizedURL;
```
Comment (robert-bogos) [inline libs/blocks/quiz/utils.js:513]: I think we can do a more concise version for this:
```suggestion
  const { locale } = getConfig();
  const prefix = locale?.prefix;

  return locale?.ietf !== 'en-US' && !originalURL.startsWith(`${prefix}/`)
  ? `${prefix}${originalURL}`
  : originalURL;
```

**Review by jedjedjedM — APPROVED** [review]

File: libs/blocks/quiz/utils.js (hunk 1 above)
Comment (JackySun9) [reply libs/blocks/quiz/utils.js:513]: If locale.ietf not define, your code cannot handle it

Comment (robert-bogos) [reply libs/blocks/quiz/utils.js:513]: I see. How about this, using destructuring and being even more concise:

```suggestion
  const { locale } = getConfig();
  const { prefix, ietf = 'en-US' } = locale || {};
  return ietf !== 'en-US' && !originalURL.startsWith(`${prefix}/`) ? `${prefix}${originalURL}` : originalURL;
```

**Comment by JackySun9** [comment]
@robert-bogos thanks for your all great feedbacks, I updated the code, please check again

**Review by robert-bogos — APPROVED** [review]

**Comment by robert-bogos** [comment]
@JackySun9 Looks great 👍 Thank you for implementing the feedback

**Review by fullcolorcoder — APPROVED** [review]

**Pull request merged**